pkg/sql/create.go

```diff
@@ -947,6 +947,12 @@ func resolveFK(
 			}
 		}
 		if !found {
+			// Avoid unexpected index builds from ALTER TABLE ADD CONSTRAINT.
+			if mode == sqlbase.ConstraintValidity_Unvalidated {
+				return errors.Errorf(
+					"foreign key columns %s must be the prefix of an index", colNames(srcCols),
```

Comment:
Sorry, that wasn't very clear. What I meant to say is that this error message seems too low-level. If I were a user who is trying to add an FK column that doesn't have a corresponding index yet, seeing this message would not help me figure out how to resolve my problem.

How about something along the lines of `cannot add foreign key columns %s without a pre-existing index`?